### ISSUE TITLE
`get_search_query()` is potentially not auto escaped, needs special handling

### ISSUE THREAD

**Issue by johnbillion** [WordPress/WordPress-Coding-Standards#1354]
The `get_search_query()` function is a special snowflake because it accepts an `$escaped` parameter which controls whether or not its output is auto escaped.

The following code is unsafe but does not get flagged by WPCS. It is potentially very easy to miss during a code review.

```php
echo '<input type="search" value="' . get_search_query( false ) . '">';
```

I think this function needs special handling which flags its use if the `$escaped` parameter is set to false.
